apps/ogrtindex.cpp

```diff
@@ -120,132 +67,168 @@ static void Usage(bool bIsError)
 MAIN_START(nArgc, papszArgv)
 
 {
+
     // Check strict compilation and runtime library version as we use C++ API.
     if (!GDAL_CHECK_VERSION(papszArgv[0]))
-        exit(1);
-    /* -------------------------------------------------------------------- */
-    /*      Register format(s).                                             */
-    /* -------------------------------------------------------------------- */
-    OGRRegisterAll();
+        GDALExit(1);
+
+    EarlySetConfigOptions(nArgc, papszArgv);
 
     /* -------------------------------------------------------------------- */
     /*      Processing command line arguments.                              */
     /* -------------------------------------------------------------------- */
-    int nFirstSourceDataset = -1;
     bool bLayersWildcarded = true;
-    const char *pszFormat = nullptr;
-    const char *pszTileIndexField = "LOCATION";
-    const char *pszOutputName = nullptr;
-    bool write_absolute_path = false;
-    bool skip_different_projection = false;
-    char *current_path = nullptr;
-    bool accept_different_schemas = false;
+    std::string osOutputFormat;
+    std::string osTileIndexField;
+    std::string osOutputName;
+    bool bWriteAbsolutePath{false};
+    bool bSkipDifferentProjection{false};
+    char *pszCurrentPath = nullptr;
+    bool bAcceptDifferentSchemas{false};
     bool bFirstWarningForNonMatchingAttributes = true;
-    const char *pszTargetSRS = "";
+    std::string osTargetSRS;
     bool bSetTargetSRS = false;
-    const char *pszSrcSRSName = nullptr;
+    std::string osSrcSRSName;
     int i_SrcSRSName = -1;
-    bool bSrcSRSFormatSpecified = false;
     SrcSRSFormat eSrcSRSFormat = FORMAT_AUTO;
     size_t nMaxFieldSize = 254;
+    std::vector<std::string> aosSrcDatasets;
+    std::vector<std::string> aosLayerNames;
+    std::vector<int> anLayerNumbers;
+
+    GDALArgumentParser argParser{"ogrtindex", true};
+
+    argParser.add_description("Program to generate a UMN MapServer compatible "
```

Comment:
Dummy``_()`` gettext macro missing here and below

Reply:
done